components/lcd/esp_lcd_ili9341/include/esp_lcd_ili9341.h

```diff
@@ -57,13 +57,25 @@ typedef struct {
  */
 esp_err_t esp_lcd_new_panel_ili9341(const esp_lcd_panel_io_handle_t io, const esp_lcd_panel_dev_config_t *panel_dev_config, esp_lcd_panel_handle_t *ret_panel);
 
+#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(4, 4, 2)
 /**
  * @brief LCD panel bus configuration structure
  *
  * @param[in] sclk SPI clock pin number
  * @param[in] mosi SPI MOSI pin number
  *
  */
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 4)
+#define ILI9341_PANEL_BUS_SPI_CONFIG(sclk, mosi)        \
+    {                                                   \
+        .sclk_io_num = sclk,                            \
+        .mosi_io_num = mosi,                            \
+        .miso_io_num = -1,                              \
+        .quadhd_io_num = -1,                            \
+        .quadwp_io_num = -1,                            \
+        .max_transfer_sz = SPI_LL_DATA_MAX_BIT_LEN >> 3,\
```

Comment:
Please, did you try on IDF4.4? The `SPI_LL_DATA_MAX_BIT_LEN` not exists in IDF4.4.  I think, that we shouldn't use this constant.

Reply:
You're right. So I added the following check before this macro:
```c
#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(4, 4, 2)
```

Reply:
Oh, I see. But I am trying on the latest IDF4.4.6 and it is not working there. Compilation failed.

Reply:
Oh my god, I tested it from v4.4 to v4.4.5, but only missed v4.4.6. I'll just make it available only after v5.0.4.

Reply:
What about to put there `#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 0)` ?
And I don't see diferencies after condition: `#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 4)`

Reply:
My recommendation:
No ifdefs and only one for all versions:
```
#define ILI9341_PANEL_BUS_SPI_CONFIG(sclk, mosi, max_transfer_sz)        \
    {                                                   \
        .sclk_io_num = sclk,                            \
        .mosi_io_num = mosi,                            \
        .miso_io_num = -1,                              \
        .quadhd_io_num = -1,                            \
        .quadwp_io_num = -1,                            \
        .max_transfer_sz = max_transfer_sz,\
    }
```

I would like to remove this include: `#include "hal/spi_ll.h"`

Reply:
> What about to put there `#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 0)` ? And I don't see diferencies after condition: `#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 4)`

In my test, the `SPI_LL_DATA_MAX_BIT_LEN ` exists in v5.0 ~ v5.0.3, and it is replaced by `SPI_LL_DMA_MAX_BIT_LEN` since v5.0.4.

Reply:
> My recommendation: No ifdefs and only one for all versions:
> 
> ```
> #define ILI9341_PANEL_BUS_SPI_CONFIG(sclk, mosi, max_transfer_sz)        \
>     {                                                   \
>         .sclk_io_num = sclk,                            \
>         .mosi_io_num = mosi,                            \
>         .miso_io_num = -1,                              \
>         .quadhd_io_num = -1,                            \
>         .quadwp_io_num = -1,                            \
>         .max_transfer_sz = max_transfer_sz,\
>     }
> ```
> 
> I would like to remove this include: `#include "hal/spi_ll.h"`

Yeah, it would be more easy. However, this macro is designed to minimize the number of input parameters, so that users don't need to worry about unnecessary configurations. And I think it can also remind users the `max_transfer_sz ` is limited by `SPI_LL_DMA_MAX_BIT_LEN`.

Reply:
I understand, but the maximum can be different by this: https://github.com/espressif/esp-idf/blob/b4268c874a4cf8fcf7c0c4153cffb76ad2ddda4e/components/driver/spi/gpspi/spi_master.c#L1190 for DMA an non-DMA. 
As I remember right, I wanted to use this macro (`SPI_LL_DMA_MAX_BIT_LEN`) in past and here is the conversation with morris: https://gitlab.espressif.cn:6688/espressif/esp-idf/-/merge_requests/23199#note_1077058

Reply:
Get it. I'll use the ``#define ILI9341_PANEL_BUS_SPI_CONFIG(sclk, mosi, max_transfer_sz) `` way. Thanks for your notice!